app/containers/List/ListPicker.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import { Text } from 'react-native';
+
+import { TActionSheetOptionsItem, useActionSheet } from '../ActionSheet';
+import { CustomIcon } from '../CustomIcon';
+import * as List from './index';
+import I18n from '../../i18n';
+import { useTheme } from '../../theme';
+import sharedStyles from '../../views/Styles';
+
+type TOption = { label: string; value: string; second?: string };
+type TOptions = { [key: string]: TOption[] };
+
+interface IListPicker {
+	title: string;
+	testID: string;
+	value?: string;
+	options: TOptions;
+	preference: string;
+	onChangeValue: (pref: string, param: { [key: string]: string }, onError: () => void) => void;
+}
+
+const ListPicker = ({ preference, options, value, title, testID, onChangeValue }: IListPicker) => {
+	const { showActionSheet, hideActionSheet } = useActionSheet();
+	const { colors } = useTheme();
+
+	const pref = value ? options[preference].find(option => option.value === value) : options[preference][0];
+
+	const [option, setOption] = useState(pref);
+
+	const actionSheetOptions: TActionSheetOptionsItem[] = options[preference].map(i => ({
+		title: I18n.t(i.label, { defaultValue: i.label, second: i.second }),
+		onPress: () => {
+			hideActionSheet();
+			onChangeValue(preference, { [preference]: i.value.toString() }, () => setOption(option));
+			setOption(i);
+		},
+		right: option?.value === i.value ? () => <CustomIcon name={'check'} size={20} color={colors.tintActive} /> : undefined
+	}));
```

Comment:
Besides this whole logic being hard to be reused because it makes use of  a `preference` context, you're re-executing the action sheet logic even when it's not needed (action sheet is closed).
Instead of that, you can create a `getOptions` function and call it when it's actually required. Example: https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/17be449d4e81649ba2025a230b7a523e248e41be/app/containers/MessageActions/index.tsx#L459

```tsx
onPress={() => showActionSheet({ options: getOptions() })}
```